kotlinx-coroutines-core/common/src/sync/Mutex.kt

```diff
@@ -352,31 +369,37 @@ internal class MutexImpl(locked: Boolean) : Mutex, SelectClause2<Any?, Mutex> {
     private abstract inner class LockWaiter(
         @JvmField val owner: Any?
     ) : LockFreeLinkedListNode(), DisposableHandle {
+        private val isTaken = atomic<Boolean>(false)
```

Comment:
it's possible to avoid this field and use an atomic wrapper over the owner (then it has to be passed between `tryResumeLockWaiter` and `completeResumeLockWaiter`), but I found it way too harsh